### PR TITLE
refactor: adapting wallet facade calls to use new method signatures and returns

### PR DESCRIPTION
**Pull request by pedroferreira1**
TODO

- [x] Stil need to bump wallet lib version to work

### CODE REVIEW

File: src/api-docs.js [hunk 1]
```diff
@@ -392,23 +392,8 @@ const apiDoc = {
...
-                        type: 'string',
-                        description: 'Name of the custom token to send the transaction.'
-                      },
-                      symbol: {
-                        type: 'string',
-                        description: 'Symbol of the custom token to send the transaction.'
-                      },
-                    }
+                    type: 'string',
+                    description: 'Token of the output. If not sent, HTR will be chosen.'
```
Comment (msbrogli) [inline src/api-docs.js:396]: Suggestion: `'Token id of the output...'`.

File: src/api-docs.js [hunk 2]
```diff
@@ -502,6 +484,10 @@ const apiDoc = {
                           type: 'integer',
                           description: 'The value parameter must be an integer with the value in cents, i.e., 123 means 1.23 HTR.'
                         },
+                        token: {
+                          type: 'string',
+                          description: 'Token of the output. If not sent, HTR will be chosen.'
```
Comment (msbrogli) [inline src/api-docs.js:489]: Same here.

File: src/index.js [hunk 3]
```diff
@@ -258,7 +258,7 @@ walletRouter.get(
   const wallet = req.wallet;
   // Expects token uid
   const token = req.query.token || null;
-  const balance = wallet.getBalance(token);
+  const balance = wallet._getBalanceRaw(token);
```
Comment (msbrogli) [inline src/index.js:261]: It seems dangerous to use internal methods that we can freely change without considering compatibility. Besides that, I feel that not all implementation of the facade have this method available.

File: src/index.js [hunk 4]
```diff
@@ -316,7 +317,7 @@ walletRouter.get('/address-index',
 walletRouter.get('/addresses', (req, res) => {
   const wallet = req.wallet;
   // TODO Add pagination
-  const addresses = wallet.getAllAddresses();
+  const addresses = wallet._getAllAddressesRaw();
```
Comment (msbrogli) [inline src/index.js:320]: Same here.

File: src/index.js [hunk 5]
```diff
@@ -334,7 +335,7 @@ walletRouter.get('/tx-history',
   }
   const wallet = req.wallet;
   const limit = req.query.limit || null;
-  const history = wallet.getTxHistory();
+  const history = wallet._getHistoryRaw();
```
Comment (msbrogli) [inline src/index.js:338]: Same here.

File: src/index.js [hunk 6]
```diff
@@ -586,36 +569,14 @@ walletRouter.post('/send-tx',
       isString: true,
       optional: true
     },
-    token: {
```
Comment (msbrogli) [inline src/index.js:589]: I like this change, but we are changing the API of the `/send-tx`. It means all integrations using it must adapt their code. Should we create a new version of the API and mark this one as deprecated?

File: src/index.js [hunk 7]
```diff
@@ -396,28 +397,11 @@ walletRouter.post('/simple-send-tx',
     },
     token: {
       in: ['body'],
-      isObject: true,
+      isString: true,
       optional: true,
-      custom: {
```
Comment (msbrogli) [inline src/index.js:401]: I like this change, but we are changing the API of the `/simple-send-tx`. It means all integrations using it must adapt their code. Should we create a new version of the API and mark this one as deprecated?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 6 above)
Comment (pedroferreira1) [reply src/index.js:589]: Do you mean to create `/send-tx-new`? Or `/v2/send-tx`?

I agree that the integrations must adapt their code but it's only if they want to upgrade the headless.

I would need to map the old parameters to the new one and have two different APIs. I guess it's fair if we decide that it's important to do it.

File: src/api-docs.js [hunk 8]
```diff
@@ -392,23 +392,8 @@ const apiDoc = {
...
-                        type: 'string',
-                        description: 'Name of the custom token to send the transaction.'
-                      },
-                      symbol: {
-                        type: 'string',
-                        description: 'Symbol of the custom token to send the transaction.'
-                      },
-                    }
+                    type: 'string | object',
+                    description: 'Token to send the transaction, just in case is not HTR. This parameter accepts a string with the token_uid (preferably) or the old format which is the full token object with {uid, name, symbol}. Support for the full token object is deprecated and will be removed soon.',
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/api-docs.js:396]: What do you think about this docs? This was the best way I found to have both parameters and I think it's very clear. 
image

[the remark this answers is not in the document]
Comment (msbrogli) [reply src/api-docs.js]: Great! Thanks!

**Pull request merged**